File: thanh123456_AcceptedCodeVNSPOJ/WCALC.cpp
#include <bits/stdc++.h>
using namespace std;
int b,n;
void nhap()
{
    scanf("%d %d",&b,&n);
}
bool check(int m)
{
    long long s1 = 0-m;
    long long s2 = b;
    long long s3 = m - 2*n;
    long long ts = s1 * s2 * s3;
    long long ms = 1ll * n * n;
    return (ts % ms ==0);
}
void solve()
{
    int res = 0;
    for (int i=1; i<2*n; i++)
    {
        res = (check(i) ? res+1 : res);
        //cout<<check(i)<<" "<<res<<endl;
    }
    cout<<res<<endl;
}
int main()
{
    nhap();
    solve();
}

File: thanh123456_AcceptedCodeVNSPOJ/WCALC.pas
const
    fin='';fon='';
var
    b,n:longint;
procedure nhap;
    begin
        assign(input,fin);reset(input);
        assign(output,fon);rewrite(output);
        readln(b,n);
    end;
function Check(m:longint):boolean;
    var
        ts,ms,s1,s2,s3:int64;
    begin
        s1:=0-m;
        s2:=b;
        s3:=m-2*n;
        ts:=s1*s2*s3;
        ms:=n;
        ms:=ms*ms;
        if ts mod ms=0 then exit(true)
        else exit(false);
    end;
procedure solve;
    var
        m,res:longint;
    begin
        res:=0;
        for m:=1 to 2*n-1 do
            if Check(m)=true then inc(res);
        writeln(res);
    close(input); close(output);
    end;
begin
    nhap;
    solve;
end.
